src/front-end/typescript/lib/index.tsx

```diff
@@ -180,21 +180,17 @@ export function fileBlobPath(file: Pick<FileRecord, "id">) {
   return prefixPath(`api/files/${file.id}?type=blob`);
 }
 
-/**
- * Intersperse React elements with a separator.
- *
- * @param collection
- * @param fn - callback to transform iteratee to a React element
- * @param separator - element placed between collection members
- * @returns List of React elements
- */
-export function intersperse<A>(
-  collection: A[],
-  fn: (a: A) => string | ReactElement,
-  separator: string | ReactElement
-) {
-  return map(collection, (val: A, index: number) => [
-    ...(index > 0 ? [separator] : []),
-    fn(val)
-  ]);
-}
+type IntersperseProps = <A>(props: {
+  collection: A[];
```

Comment:
I don't think there is a need for a type parameter `A` here.

`collection: ReactElement[] | string;`

Based on its current usage the TS compiler will automatically coerce the `A` to `unknown` (same problem that we had with the front-end HTTP modules).

Also, I think `IntersperseProps` should be an interface:

```ts
interface IntersperseProps {
  collection: Array<ReactElement | string>;
  separator: ReactElement | string;
}
```

Then, `Intersperse` would be:

```ts
export function Intersperse(props: IntersperseProps): ReactElement {
  // ...
}
```

Reply:
Sorry, one suggestion, you could write `Intersperse` as:

```ts
export const Intersperse: View<IntersperseProps> = ({ collection, separator }) => {
  // ...
}
```